src/utils/particle/drawer.ts

```diff
@@ -171,15 +185,25 @@ export class Drawer {
             ratio = particle.img.ratio;
         }
 
-        ctx.drawImage(img_obj, particle.position.x - radius, particle.position.y - radius, radius * 2, radius * 2 / ratio);
+        const pos = {
+            x: particle.position.x - radius,
+            y: particle.position.y - radius,
+        };
+
+        ctx.drawImage(img_obj, pos.x, pos.y, radius * 2, radius * 2 / ratio);
     }
 
-    public static subDrawShape(ctx: CanvasRenderingContext2D, startX: number, startY: number, sideLength: number, sideCountNumerator: number, sideCountDenominator: number) {
+    private static subDrawShape(ctx: CanvasRenderingContext2D,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Declaration of private static method not allowed after declaration of private instance method. Instead, this should come after constructors.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5011230)